ziti/src/main/kotlin/io/netfoundry/ziti/identity/util.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2018-2020 NetFoundry, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.netfoundry.ziti.identity
+
+import io.netfoundry.ziti.util.readCerts
+import io.netfoundry.ziti.util.readKey
+import org.bouncycastle.asn1.x500.X500Name
+import org.bouncycastle.asn1.x500.style.BCStyle
+import java.io.File
+import java.net.URI
+import java.security.KeyStore
+
+/**
+ *
+ */
+internal fun findIdentityAlias(ks: KeyStore): String {
+    for (a in ks.aliases()) {
+        if (ks.isKeyEntry(a)) {
+            return a
+        }
+    }
+
+    error("no suitable key entry")
+}
+
+internal fun keystoreFromConfig(id: IdentityConfig): KeyStore {
+    val ks = KeyStore.getInstance("PKCS12")
+    ks.load(null)
+
+    val certs = readCerts(id.id.cert.replace("pem:", ""))
+    val ztAPI = URI.create(id.ztAPI)
+    val name = X500Name(certs[0].subjectDN.name).getRDNs(BCStyle.CN)[0].first.value.toASN1Primitive().toString()
```

Comment:
position 0 is guaranteed to not be null i assume. also - is position 0 guaranteed to be 'the right' one?

Reply:
there is only one, just API is overly generic